nle/env/base.py

```diff
@@ -379,15 +403,18 @@ def step(self, action: int):
         last_observation = tuple(a.copy() for a in self.last_observation)
 
         observation, done = self.env.step(self._actions[action])
-        observation, done = self._perform_known_steps(
-            observation, done, exceptions=True
-        )
+        if (
+            not self._allow_all_modes or observation[self._program_state_index][0] == 1
```

Comment:
What does this do? Doesn't seem very readable to me.

Reply:
This checks if death the game is over. will improve clarity